Scripts/STRAXSidechainRegistrationScript.ps1

```diff
@@ -0,0 +1,593 @@
+                {
+                    $a = Get-LocalHeight
+                    $b = Get-WalletHeight 
+                    $c = $a - $b
+                    ""
+                    Write-Host (Get-TimeStamp) "The Wallet Synced Height is $b" -ForegroundColor Yellow
+                    Write-Host (Get-TimeStamp) "The Current Tip is $a" -ForegroundColor Yellow
+                    Write-Host (Get-TimeStamp) "$c Blocks are Required..." -ForegroundColor Yellow
+                    Start-Sleep 10
+                }
+            }
+
+            No
+            {
+                Write-Host (Get-TimeStamp) "ERROR: There was some missing wallet detail - Please re-run this script" -ForegroundColor Red
+                Start-Sleep 30
+                Exit
+            }
+        }
+    }
+    
+
+#Check Wallet Balance
+$CirrusWalletBalance = (Invoke-WebRequest -Uri http://localhost:$API/api/Wallet/balance?WalletName=$Wallet -Method Get | Select-Object -ExpandProperty content | ConvertFrom-Json | Select-Object -ExpandProperty balances | Select-Object -ExpandProperty spendableamount) / 100000000
+if ( $CirrusWalletBalance -ge 500.01 )
+{
+    Write-Host (Get-TimeStamp) "SUCCESS: $Wallet contains a balance of over 501 CRS!" -ForegroundColor Green
+}
+    Else
+    {
+        Write-Host (Get-TimeStamp) "ERROR: $Wallet does not contain a balance of over 501 CRS! Please run again and define a wallet that contains the required amount..." -ForegroundColor Red
+        Start-Sleep 30
+        Exit
+    }
+
+#Gather Federation Detail
+""
+if ( -not ( Test-Path $sideChainDataDir\federationKey.dat ) ) 
+{
+    $miningDAT = Read-Host "Please Enter the full path to the federationKey.dat"
+    While ( -not ( Test-Path $miningDAT ) )
+    {
+        Write-Host Write-Host (Get-TimeStamp) "ERROR: $miningDAT is not a valid path.. Be sure to include the filename too '\federationKey.dat'" -ForegroundColor Red
+        $miningDAT = Read-Host "Please Enter the full path to the federationKey.dat"
+    }
+    Copy-Item $miningDAT -Destination $sideChainDataDir -Force -ErrorAction Stop
+}
+
+
+#Perform Registration
+""
+$CollateralAddress = Read-Host -Prompt "Please enter your STRAX Address that contains the required collateral amount"
+""
+$RegisterMasternode = Read-Host -Prompt 'Would you like to register as a Masternode? Please be aware that this will incur a 500 CRS Fee. Enter "Yes" to continue or "No" to exit the script'
+While ( $RegisterMasternode -ne "Yes" -and $RegisterMasternode -ne "No" )
+{
+    ""
+    $RegisterMasternode = Read-Host 'Enter "Yes" to continue or "No" to exit the script'
+    ""
+}
+Switch ( $RegisterMasternode )
+{
+    Yes
+    {
+        $Body = @{}
+        $Body.Add("collateralAddress",$CollateralAddress)
+        $Body.Add("collateralWalletName",$CollateralWallet)
+        $Body.Add("collateralWalletPassword",$CollateralWalletPassword)
+        $Body.Add("walletName",$CirrusWallet)
+        $Body.Add("walletPassword",$WalletPassword)
```

Comment:
`$WalletPassword` might hold the Collateral Wallet Password if that was restored. It might hold the Cirrus Wallet Password if that wallet was restored. Or it might not yet have been initiated when neither wallet needed to be restored.

Reply:
Thanks for your review - Sorted!